### PR TITLE
Introduce a workflow to automate the build-time stats generation for samples

### PR DESCRIPTION
**Pull request by Thevakumar-Luheerathan**
- Clone nBallerina, Hello_world, Hello_world_service samples from ballerina-lang repo.
- Set-up Ballerina in the runner.
- Build each project with -dump-build-time
- Remove old stats
- Push back the new stats as bot.

### CODE REVIEW

File: .github/workflows/generate-build-time-stats.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: Generate build time statistics 
+
+
+on: 
+  workflow_dispatch:
+  schedule:
+    - cron: '*/5 * * * *'
```
Comment (azinneera) [inline .github/workflows/generate-build-time-stats.yml:7]: Shall we change this to once a day?

File: .github/workflows/generate-build-time-stats.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      
+      - name: Push changes to remote
+        run: |
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+
+          git add build-time-data
+          git commit -m "Update build time stats"
+          git push
+
```
Comment (azinneera) [inline .github/workflows/generate-build-time-stats.yml:38]: Let's remove these unnecessary new lines

**Comment by azinneera** [comment]
The PR heading should use the imperative mood for the PR titles and commit messages. i.e: `Introduce a workflow..` instead of `Introducing a workflow...`. The commit messages are correct. Follow the same format for PR title as well.

File: .github/workflows/generate-build-time-stats.yml (hunk 1 above)
Comment (Thevakumar-Luheerathan) [reply .github/workflows/generate-build-time-stats.yml:7]: ok.Sure I will

File: .github/workflows/generate-build-time-stats.yml (hunk 2 above)
Comment (Thevakumar-Luheerathan) [reply .github/workflows/generate-build-time-stats.yml:38]: ok sure..

File: .github/workflows/generate-build-time-stats.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+          cp build-time.json ./build-time-data/hello_world_service.json
+
+      - name: Push changes to remote
+        run: |
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+
+          git add build-time-data
+          git commit -m "Update build time stats"
+          git push
```
Comment (azinneera) [inline .github/workflows/generate-build-time-stats.yml:64]: Missing new line

File: .github/workflows/generate-build-time-stats.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
+name: Generate build time statistics 
+
```
Comment (azinneera) [inline .github/workflows/generate-build-time-stats.yml:2]: Unnecessary new line

**Review by azinneera — APPROVED** [review]

**Pull request merged**